### ISSUE TITLE
Making switch state more noticeable

### ISSUE THREAD

**Issue by muratcorlu** [Trendyol/grace#537]
For "switch component", with arbitrary colors, it's not very clear if it's "on" or "off". For example, when I see switch with `danger` color, by alone, I can not immediately understand if it's "off" or "on". Sometimes opening something can be also dangerous, right? 😊 

I thought there should be something else other than just color that indicates the state of the switch. I noticed iOS has an option like below in accessibility settings:

![IMG_9771](https://user-images.githubusercontent.com/127687/232079802-97a50f15-8575-43fd-9b7f-11de72e2c57e.jpg)

And it may be nice to notice that, in iOS, switches are always green and gray. So state/color relation is much more obvious. But even in that case they add this option in accessibility settings. Which gives me the idea that, our switch with different colors than default colors has a big risk to be non accessible.

Maybe we can also consider a mode like that for our switch. This can be optional or automatically by respecting some related OS level settings (like `prefers-contrast`).

_Originally discussion started in #523_
